hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntitySelectFetchInitializerBuilder.java

```diff
@@ -83,7 +83,7 @@ private static BatchMode determineBatchMode(
 			EntityPersister entityPersister,
 			FetchParentAccess parentAccess,
 			AssemblerCreationState creationState) {
-		if ( !entityPersister.isBatchLoadable() || creationState.isScrollResult() ) {
+		if ( !entityPersister.isBatchLoadable() || creationState.disableBatching() ) {
```

Comment:
Wouldn't it be better to use `BatchMode.BATCH_INITIALIZE` if there is a dynamic instantiation referring to the fetch owner (i.e. the `FetchParentAccess` root)?

Reply:
I did a quick test and `BATCH_INITIALIZE` would work in this very simple case; I did test and verify that in 5 we disabled batching for dynamic instantiation queries so I thought that doing the same would be safest.

Sadly we can't determine if the `FetchParentAccess` root is a dynamic instantiation because when the entity result initializer is created from a `DynamicInstantiationResult`, the latter is a first level `DomainResult` so the fetch parent access is always set to `null`.

Reply:
Ok, then this is fine. Let's not overcomplicate this.

Reply:
Actually, I think `BATCH_INITIALIZE` is fine here, it will always eagerly load any associations and that's the same we did in 5. We also use it for cacheable entities, so I don't see why it wouldn't work here.

I'll also cache the check for dynamic instantiations so we don't have to repeat it.